sources/code/main/modules/extensions.ts

```diff
@@ -31,32 +38,30 @@ async function parseImports(cssString: string, importCalls:string[], maxTries=5)
       promises.push(Promise.reject(new Error("Circular reference in CSS imports are disallowed!")));
       return;
     }
-    importCalls.push(file);
-    promises.push(fetchOrRead(file)
+    promises.push(fetchOrRead(file, prevImport)
       .then(data => {
         if (data.download)
           return data.download.then(data => data.text());
         else
           return data.read.then(data => data.toString());
       })
+      .then(content => parseImports(content, [...importCalls, file], maxTries, dirname(file)))
       .then(content => cssString = cssString.replace(singleImport, content))
     );
   });
   try {
     await Promise.all(promises);
   } catch(error) {
     if(maxTries > 0) {
-      console.warn("Couldn't resolve CSS theme imports, retrying again...");
-      maxTries--;
+      console.error(error);
+      console.warn("Couldn't resolve CSS theme imports, retrying...");
+      return parseImports(cssString, importCalls, maxTries - 1, prevImport);
     }
     else if(error instanceof Error)
       throw error;
     else
       throw new Error("Couldn't resolve CSS theme imports, aborting...");
   }
-  if(anyImport.test(cssString)) {
-    return parseImports(cssString, importCalls, maxTries);
-  }
```

Comment:
I'd leave that untouched and prepare the code, just so there's only one place a `parseImports` is called. This keeps code to function and look a bit more as a recursion is done linearly, and a single file is parsed during the single call. If you think of that, I've kinda coded it so there's *as little as it can be of the recursion* in this part of the code.  But it is necessary since we need to resolve the imports first in order to estimate the next ones.

Reply:
Done